### PR TITLE
[docker] move to ubuntu 20.04

### PR DESCRIPTION
**Pull request by gjc13**

### CODE REVIEW

**Comment by simonlingoogle** [comment]
Did you measure the impact on docker image size for upgrading Ubuntu?

File: etc/docker/Dockerfile [hunk 1]
```diff
@@ -71,6 +71,7 @@ ENV OTBR_NORELEASE_DEPS \
 
 RUN apt-get update \
   && apt-get install --no-install-recommends -y $OTBR_DOCKER_REQS $OTBR_DOCKER_DEPS \
+  && [ "${OT_BACKBONE_CI}" != "1" ] || apt-get install --no-install-recommends -y $OTBR_OT_BACKBONE_CI_DEPS \
```
Comment (bukepo) [inline etc/docker/Dockerfile:74]:
```suggestion
  && ([ "${OT_BACKBONE_CI}" != "1" ] || apt-get install --no-install-recommends -y $OTBR_OT_BACKBONE_CI_DEPS) \
```

**Review by simonlingoogle — APPROVED** [review]
LGTM 👍

**Review by bukepo — APPROVED** [review]
LGTM:+1:

**Review by Irving-cl — APPROVED** [review]
LGTM

**Review by jwhui — APPROVED** [review]

**Pull request merged**